### PR TITLE
fix(ThreadListViewModel): Throttle sections update

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

File: Mail/Views/Thread List/ThreadListViewModel+Observation.swift [hunk 1]
```diff
@@ -55,10 +55,8 @@ extension ThreadListViewModel {
 
                     DispatchQueue.main.sync {
                         self.filteredThreads = filteredThreads
-                        withAnimation(animateInitialThreadChanges ? .default : nil) {
-                            self.sections = newSections
-                        }
                     }
+                    sectionsSubject.send(newSections)
```
Comment (PhilippeWeidmann) [inline Mail/Views/Thread List/ThreadListViewModel+Observation.swift:59]: Technically this is not the same as before as the call is now async and throttled but the I didn't see issues with my tests.

File: Mail/Views/Thread List/ThreadListViewModel+Observation.swift [hunk 2]
```diff
@@ -41,7 +41,7 @@ extension ThreadListViewModel {
 
     // MARK: - Observe global changes
 
-    func observeChanges(animateInitialThreadChanges: Bool = false) {
+    func observeChanges() {
```
Comment (PhilippeWeidmann) [inline Mail/Views/Thread List/ThreadListViewModel+Observation.swift:44]: Now we always perform an animation. I seems like doesn't have any side effect even at initial launch.

File: Mail/Views/Thread List/ThreadListViewModel.swift [hunk 3]
```diff
@@ -182,6 +187,13 @@ final class DateSection: Identifiable, Equatable {
         self.mailboxManager = mailboxManager
         self.folder = folder
         self.isCompact = isCompact
+        sectionsObserver = sectionsSubject
+            .throttle(for: .seconds(1), scheduler: DispatchQueue.main, latest: true)
```
Comment (PhilippeWeidmann) [inline Mail/Views/Thread List/ThreadListViewModel.swift:191]: I put a 1 sec throttle. Maybe it's too much, didn't see any issue in my tests.

**Review by adrien-coye — APPROVED** [review]
Looks good to me, I'll test it in beta.

**Review by valentinperignon — APPROVED** [review]
seems amazing

**Pull request merged**